test/services/resolveCourtCase.integration.test.ts

```diff
@@ -0,0 +1,386 @@
+      expect(afterCourtCase.errorResolvedBy).toEqual(resolverUsername)
+      expect(afterCourtCase.errorResolvedTimestamp).not.toBeNull()
+      expect(afterCourtCase.notes[0].userId).toEqual("System")
+      expect(afterCourtCase.notes[0].noteText).toEqual(
+        `${resolverUsername}: Portal Action: Record Manually Resolved.` +
+          ` Reason: ${resolution.reason}. Reason Text: ${resolution.reasonText}`
+      )
+
+      expect(afterCourtCase.resolutionTimestamp).toBeNull()
+    })
+  })
+
+  describe("when there is an unexpected error", () => {
+    const resolution: ManualResolution = {
+      reason: "NonRecordable"
+    }
+
+    beforeEach(async () => {
+      await insertCourtCasesWithFields([
+        {
+          errorLockedByUsername: resolverUsername,
+          triggerLockedByUsername: resolverUsername,
+          orgForPoliceFilter: visibleForce
+        }
+      ])
+    })
+
+    it("should return the error if fails to create notes", async () => {
+      ;(insertNotes as jest.Mock).mockImplementationOnce(() => new Error(`Error while creating notes`))
+
+      let result: UpdateResult | Error
+      try {
+        result = await resolveCourtCase(dataSource, 0, resolution, user)
+      } catch (error) {
+        result = error as Error
+      }
+      expect(result).toEqual(Error(`Error while creating notes`))
+
+      const record = await dataSource.getRepository(CourtCase).findOne({ where: { errorId: 0 } })
+      const actualCourtCase = record as CourtCase
+
+      expectToBeUnresolved(actualCourtCase)
+    })
+
+    it("should return the error if fails to unlock the case", async () => {
+      ;(unlockCourtCase as jest.Mock).mockImplementationOnce(() => new Error(`Error while unlocking the case`))
+
+      let result: UpdateResult | Error
+      try {
+        result = await resolveCourtCase(dataSource, 0, resolution, user)
+      } catch (error) {
+        result = error as Error
+      }
+
+      expect(result).toEqual(Error(`Error while unlocking the case`))
+
+      const record = await dataSource.getRepository(CourtCase).findOne({ where: { errorId: 0 } })
+      const actualCourtCase = record as CourtCase
+
+      expectToBeUnresolved(actualCourtCase)
+    })
+
+    it("should return the error when fails to update the case", async () => {
+      jest
+        .spyOn(UpdateQueryBuilder.prototype, "execute")
+        .mockRejectedValue(Error("Failed to update record with some error"))
+
+      let result: UpdateResult | Error
+      try {
+        result = await resolveCourtCase(dataSource, 0, resolution, user)
+      } catch (error) {
+        result = error as Error
+      }
+      expect(result).toEqual(Error(`Failed to update record with some error`))
+
+      const record = await dataSource.getRepository(CourtCase).findOne({ where: { errorId: 0 } })
+      const actualCourtCase = record as CourtCase
+
+      expectToBeUnresolved(actualCourtCase)
+    })
+  })
+})
```

Comment:
Should there be a block that describes the situation in which there are no exceptions on a case, only triggers?